nav2_simple_commander/nav2_simple_commander/robot_navigator.py

```diff
@@ -46,50 +46,63 @@ class TaskResult(Enum):
+    def __init__(self, node_name='basic_navigator', namespace=''):
         super().__init__(node_name=node_name)
         self.initial_pose = PoseStamped()
         self.initial_pose.header.frame_id = 'map'
         self.goal_handle = None
         self.result_future = None
         self.feedback = None
         self.status = None
+        self.namespace = namespace
 
         amcl_pose_qos = QoSProfile(
-          durability=QoSDurabilityPolicy.TRANSIENT_LOCAL,
-          reliability=QoSReliabilityPolicy.RELIABLE,
-          history=QoSHistoryPolicy.KEEP_LAST,
-          depth=1)
+            durability=QoSDurabilityPolicy.TRANSIENT_LOCAL,
+            reliability=QoSReliabilityPolicy.RELIABLE,
+            history=QoSHistoryPolicy.KEEP_LAST,
+            depth=1)
 
         self.initial_pose_received = False
         self.nav_through_poses_client = ActionClient(self,
                                                      NavigateThroughPoses,
-                                                     'navigate_through_poses')
-        self.nav_to_pose_client = ActionClient(self, NavigateToPose, 'navigate_to_pose')
-        self.follow_waypoints_client = ActionClient(self, FollowWaypoints, 'follow_waypoints')
-        self.follow_path_client = ActionClient(self, FollowPath, 'follow_path')
-        self.compute_path_to_pose_client = ActionClient(self, ComputePathToPose,
-                                                        'compute_path_to_pose')
-        self.compute_path_through_poses_client = ActionClient(self, ComputePathThroughPoses,
-                                                              'compute_path_through_poses')
+                                                     f'/{self.namespace}/navigate_through_poses')
+        self.nav_to_pose_client = ActionClient(self,
+                                               NavigateToPose,
+                                               f'/{self.namespace}/navigate_to_pose')
+        self.follow_waypoints_client = ActionClient(self,
+                                                    FollowWaypoints,
+                                                    f'/{self.namespace}/follow_waypoints')
+        self.follow_path_client = ActionClient(self,
+                                               FollowPath,
+                                               f'/{self.namespace}/follow_path')
+        self.compute_path_to_pose_client = ActionClient(self,
+                                                        ComputePathToPose,
+                                                        f'/{self.namespace}/compute_path_to_pose')
+        self.compute_path_through_poses_client = ActionClient(self,
+                                                              ComputePathThroughPoses,
+                                                              f'/{self.namespace}/compute_path_through_poses')
         self.smoother_client = ActionClient(self, SmoothPath, 'smooth_path')
```

Comment:
Missed these 3?

Reply:
Yes, I agree that all the services also need namespace added, but I did not test those and was not sure if the change would simply work.